map/local_ads_helpers.cpp

```diff
@@ -0,0 +1,92 @@
+#include "map/local_ads_helpers.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/assert.hpp"
+#include "base/bits.hpp"
+
+namespace
+{
+template<typename TPrimitive>
+TPrimitive ReadPrimitive(FileReader const & reader, uint64_t & offset)
+{
+  if (offset + sizeof(TPrimitive) > reader.Size())
+    MYTHROW(Reader::SizeException, ("File is corrupted", offset, sizeof(TPrimitive), reader.Size()));
+
+  std::vector<char> bytes(sizeof(TPrimitive));
+  ReadFromPos(reader, offset, bytes.data(), bytes.size());
+  offset += bytes.size();
+
+  MemReaderWithExceptions memReader(bytes.data(), bytes.size());
+  ReaderSource<decltype(memReader)> src(memReader);
+  return ReadPrimitiveFromSource<TPrimitive>(src);
+}
+}
+
+void WriteCountryName(FileWriter & writer, std::string const & countryName)
+{
+  ASSERT(!countryName.empty(), ());
+  ASSERT_LESS(countryName.length(), (1 << sizeof(uint8_t) * 8) - 1, ());
+  uint8_t len = static_cast<uint8_t>(countryName.length() + 1);
+  WriteToSink(writer, len);
```

Comment:
There is a reader stream that may read string (see coding/streams.hpp). The same for writer stream.